src/main/java/filter/expression/QualifierContent.java

```diff
@@ -0,0 +1,66 @@
+package filter.expression;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class QualifierContent {
+
+    //Only one of these will exist in a QualifierContent object
```

Comment:
I think we can just use the existing Qualifier class for this purpose.

Actually, can we just move the `parseSeparatedContent` part to `parseQualifier` as below? This way, it would be able to parse qualifier content which might need different parsing function to be parsed (for example, to parse `id:<5;6 .. 7;10` we need to use `parseRangeOperator` and `parseNumberOrNumberRange`).

```java
private FilterExpression parseQualifier(Token token) {
    String qualifierName = token.getValue();

    // Strip the : at the end, then trim
    qualifierName = qualifierName.substring(0, qualifierName.length() - 1).trim();

    QualifierType type = QualifierType.parse(qualifierName).orElse(QualifierType.FALSE);
    if (type == QualifierType.SORT) { //special case for sort
        return new Qualifier(type, parseSortKeys());
    } else {
        return parseSeparatedContent(type, () -> parseQualifierContent);
    }
}
```